app/mailers/user_mailer.rb

```diff
@@ -219,7 +221,9 @@ def set_instance
     @instance = Rails.configuration.x.local_domain
   end
 
-  def locale(use_current_locale: false)
-    @resource.locale.presence || (use_current_locale && I18n.locale) || I18n.default_locale
+  def with_user_settings(user, use_current_locale: false, &block)
+    I18n.with_locale(user.locale || (use_current_locale && I18n.locale) || I18n.default_locale) do
+      Time.use_zone(user.time_zone || ENV['DEFAULT_TIME_ZONE'] || 'UTC', &block)
```

Comment:
For purpose of this PR this is fine -- same future note here as the previous one on initializer though -- we should pull the second part of this out to a `Rails.configuration.x....` type value that checks the env and defaults.